tests/model_registry/rest_api/conftest.py

```diff
@@ -63,3 +90,272 @@ def updated_model_registry_resource(
+    ) as mr:
+        mr.wait_for_condition(condition="Available", status="True")
+        yield mr
+
+
+@pytest.fixture()
+def local_ca_bundle(request: pytest.FixtureRequest, admin_client: DynamicClient) -> Generator[str, Any, Any]:
+    """
+    Creates a local CA bundle file by fetching the CA bundle from a ConfigMap and appending the router CA from a Secret.
+    Args:
+        request: The pytest request object
+        admin_client: The admin client to get the CA bundle from a ConfigMap and append the router CA from a Secret.
+    Returns:
+        Generator[str, Any, Any]: A generator that yields the CA bundle path.
+    """
+    ca_bundle_path = getattr(request, "param", {}).get("cert_name", "ca-bundle.crt")
+    create_ca_bundle_with_router_cert(
+        client=admin_client,
+        namespace=py_config["model_registry_namespace"],
+        ca_bundle_path=ca_bundle_path,
+        cert_name=ca_bundle_path,
+    )
+    yield ca_bundle_path
+
+    os.remove(ca_bundle_path)
+
+
+@pytest.fixture(scope="class")
+def ca_configmap_for_test(
+    admin_client: DynamicClient,
+    model_registry_namespace: str,
+    mysql_ssl_artifact_paths: dict[str, Any],
+) -> Generator[ConfigMap, None, None]:
+    """
+    Creates a test-specific ConfigMap for the CA bundle, using the generated CA cert.
+
+    Args:
+        admin_client: The admin client to create the ConfigMap
+        model_registry_namespace: The namespace of the model registry
+        mysql_ssl_secrets: The artifacts and secrets for the MySQL SSL connection
+
+    Returns:
+        Generator[ConfigMap, None, None]: A generator that yields the ConfigMap instance.
+    """
+    with open(mysql_ssl_artifact_paths["ca_crt"], "r") as f:
+        ca_content = f.read()
+    if not ca_content:
+        LOGGER.info("CA content is empty")
+        raise Exception("CA content is empty")
+    cm_name = "mysql-ca-configmap"
+    with ConfigMap(
+        client=admin_client,
+        name=cm_name,
+        namespace=model_registry_namespace,
+        data={"ca-bundle.crt": ca_content},
+    ) as cm:
+        yield cm
+
+
+@pytest.fixture(scope="class")
+def patch_mysql_deployment_with_ssl_ca(
+    request: pytest.FixtureRequest,
+    admin_client: DynamicClient,
+    model_registry_namespace: str,
+    model_registry_db_deployment: Deployment,
+    mysql_ssl_secrets: dict[str, Any],
+    ca_configmap_for_test: ConfigMap,
+) -> Generator[Deployment, Any, Any]:
+    """
+    Patch the MySQL deployment to use the test CA bundle (mysql-ca-configmap),
+    and mount the server cert/key for SSL.
+    """
+    CA_CONFIGMAP_NAME = request.param.get("ca_configmap_name", "mysql-ca-configmap")
+    CA_MOUNT_PATH = request.param.get("ca_mount_path", "/etc/mysql/ssl")
+    deployment = Deployment(
+        client=admin_client,
+        name=model_registry_db_deployment.name,
+        namespace=model_registry_namespace,
+    )
+    deployment.wait_for_condition(condition="Available", status="True")
```

Comment:
you already are requesting model_registry_db_deployment in this fixture, you should be able to use it here.